### PR TITLE
BinaryOperatorSpacesFixer - Fix spaces around multiple exception catching (|)

### LINKED ISSUE #3207: BinaryOperatorSpacesFixer doesn't work with multiple exceptions in one catch block
PHP version: 7.1.9
PHP CS Fixer version: 2.8.0

In PHP 7.1 multiple exceptions can be specified in a single `catch` block using the `|` operator.

The BinaryOperatorSpacesFixer doesn't fix the spaces around `|` when used in a `catch` block.

Actual result:
```php
try {
    // some code
} catch (Exception1|       Exception2 $e) {
}
```

Expected result:
```php
try {
    // some code
} catch (Exception1 | Exception2 $e) {
}
```

### PR DESCRIPTION
**Pull request by ntzm**
Continued from #3210, sorry for the new PR, couldn't figure out how to rebase it to 2.2. Closes #3207

### CODE REVIEW

File: src/Tokenizer/TokensAnalyzer.php [hunk 1]
```diff
@@ -461,6 +461,7 @@ public function isBinaryOperator($index)
                 T_SR => true,                   // >>
                 T_SR_EQUAL => true,             // >>=
                 T_XOR_EQUAL => true,            // ^=
+                CT::T_TYPE_ALTERNATION => true, // |
```
Comment (ntzm) [inline src/Tokenizer/TokensAnalyzer.php]: Should this be added to the array conditionally? Because it's a PHP 7.1 only feature

Comment (SpacePossum) [reply src/Tokenizer/TokensAnalyzer.php]: should be good here, thank for .7 target PR btw :)

**Comment by SpacePossum** [comment]
```

/usr/local/Homebrew/Library/Homebrew/brew.rb:12:in `<main>': Homebrew must be run under Ruby 2.3! You're running 2.0.0. (RuntimeError)

brew tap homebrew/homebrew-php returned exit code 1
```

this cannot be helped by @ntzm or me, however; @ntzm could you rebase on top of 2.2? It should also trigger a new build so maybe it passes this time

**Comment by ntzm** [comment]
Shall I close the 2.8 PR?

**Comment by SpacePossum** [comment]
>Shall I close the 2.8 PR?

If there isn't anything in that one that is in this one, you can safely close it.
Please make a link in a comment to the PR so when @keradus merges it upstream and runs into conflicts he can use it as reference.

ps. you already made the link in the description of this PR, so all good, thanks :)

**Comment by keradus** [comment]
Thank you @ntzm.

**Pull request merged**